src/panels/filter-panel-react/FilterPanelControl.tsx

```diff
@@ -1,45 +1,214 @@
+        }
+
+        return []
     }
 
-    const selectChanged = () => {
-        //TODO: When these values are real, store this selection.
+    const generateFilterSelectOptions = (filter: ActiveFilter) => {
+        const filterId = getFilterId(filter)
+        const [key, currentValues] = metricValues.find(([k,v]) => k === filterId) || []
+
+        if (key && currentValues) {
+            const values = currentValues.map(v => ({
+                label: v.text,
+                value: v.value || ''
+            } as SelectableValue<string | number>))
+
+            return values
+        }
+
+        return []
     }
 
+    const filterDisplayLabel = (filter: ActiveFilter, index: number) => {
+        const altLabel = filter.altColumnLabel
 
-    useEffect(() => {
+        if (altLabel) {
+            return altLabel
+        }
+
+        return `${filter.entity.label || ''}: ${filter.attribute.label || ''}`
+    }
+
+    const getInputSelectableValue = (filter: ActiveFilter) => {
+        const filterId = getFilterId(filter)
+        const selVals = selectableValues.find(v => v.filterId === filterId)
+
+        if (selVals && selVals.values?.length > 0) {
+            return selVals.values[0].value || ''
+        }
+
+        return ''
+    }
+
+    const getSelectSelectableValues = (filter: ActiveFilter) => {
+        const filterId = getFilterId(filter)
+        const selVals = selectableValues.find(v => v.filterId === filterId)
+
+        return (selVals && selVals.values) || []
+    }
+
+    const inputChanged = (e, filter: ActiveFilter) => {
+        const filterId = getFilterId(filter)
+        const newValues = selectableValues.filter(v => v.filterId !== filterId)
+
+        const newValue = {
+            label: '',
+            value: (e.target.value || '') as string
+        }
+
+        const newEntry = {
+            filterId,
+            values: [newValue]
+        } as FilterSelectableValues
+
+        newValues.push(newEntry)
+        setSelectableValues(newValues)
+        saveFilterPanel(newValues)
+    }
+
+    const selectChanged = (value: (SelectableValue | SelectableValue[]), filter: ActiveFilter) => {
+        const filterId = getFilterId(filter)
+        const newValue = Array.isArray(value) ? value : [value]
+        const newValues = selectableValues.filter(v => v.filterId !== filterId)
+
+        const newEntry = {
+            filterId,
+            values: newValue
+        } as FilterSelectableValues
+
+        newValues.push(newEntry)
+        setSelectableValues(newValues)
+        saveFilterPanel(newValues)
+    }
+
+    /**
+     * Save filter panel data to localStorage, for use by Entity Datasource when filtering queries.
+     */
+    const saveFilterPanel = (filterSelectableValues: FilterSelectableValues[]) => {
         if (props.options.filterEditor) {
-            saveFilterPanel(props.options.filterEditor)
+            // TODO: May want to remove some data from datasource before saving
```

Comment:
This should definitely be tracked in a JIRA and double checked before going live, this was a quick and dirty solution and your concerns about possibly storing sensitive data is legit and should be verified.